src/Eshopworld.Core/IMessenger.cs

```diff
@@ -5,50 +5,117 @@
     using JetBrains.Annotations;
 
     /// <summary>
-    /// Contract that provides a simple way to send messages.
+    /// Contract that provides a simple way to send messages to queues.
     /// </summary>
     public interface ISendMessages : IDisposable
     {
         /// <summary>
-        /// Sends a message.
+        /// Sends a message onto a queue.
         /// </summary>
         /// <typeparam name="T">The type of the message that we are sending.</typeparam>
         /// <param name="message">The message that we are sending.</param>
-        Task Send<T>([NotNull] T message) where T : IMessage;
+        Task Send<T>([NotNull] T message) where T : class;
+    }
+
+    /// <summary>
+    /// Contract that provides a simple way to send events to topics.
+    /// </summary>
+    public interface IPublishEvents : IDisposable
```

Comment:
Also why not putting each interface into their own file? (makes for much easier solution navigation)